src/carousel/index.ts

```diff
@@ -0,0 +1,283 @@
+
+    const transformStyles = {
+      transform: tfm,
+      '-webkit-transform': tfm,
+      '-ms-transform': tfm,
+    };
+
+    const tsVal = speed + 'ms ' + 'ease';
+    const transitionStyles = transition === true ? {
+      '-webkit-transition': tsVal,
+      transition: tsVal
+    } : {};
+
+    const style = Object.assign({}, {
+      width: `${trackWidth}px`,
+    },
+      transformStyles,
+      transitionStyles);
+
+    if (this.temporaryNoTransition) {
+      delete style['transition'];
+      delete style['-webkit-transition'];
+      this.temporaryNoTransition = !this.temporaryNoTransition;
+    }
+    return style;
+  }
+
+  getDots = () => {
+    if (!this.props.items) {
+      return;
+    }
+    const slideCount = this.props.items.length;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const slidesToScroll = this.props.settings.slidesToScroll || DEFAULT_SETTINGS.slidesToScroll;
+
+    const dotCount = Math.ceil((slideCount - slidesToShow) / slidesToScroll + 1);
+
+    return Array(dotCount).fill('dot');
+  }
+
+  getDotStyle = (i: number) => {
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+    const slidesToScroll = this.props.settings.slidesToScroll || DEFAULT_SETTINGS.slidesToScroll;
+
+    let leftBound = i * slidesToScroll;
+    let rightBound = i * slidesToScroll + slidesToShow - 1;
+
+    if (this.currentSlide >= leftBound && this.currentSlide <= rightBound) {
+      return { 'background-color': 'black' };
+    } else {
+      return {};
+    }
+  }
+
+  swipeSlides = (touchObj: { startX: number, startY: number, curX: number, curY: number }) => {
+    const direction: string = calSwipeDirection(touchObj);
+
+    if (direction === 'left') {
+      this.moveNext();
+    } else if (direction === 'right') {
+      this.movePrevious();
+    }
+  }
+
+  getSlideWidth = () => {
+    const visibleWidth = this.refs.carouselwrap.offsetWidth;
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    if (visibleWidth) {
+      const slideWidth = visibleWidth / slidesToShow;
+      return slideWidth;
+    }
+  }
+
+  calcPos = (currS: number, moveDistance: number): number => {
+    const slidesToShow = this.props.settings.slidesToShow || DEFAULT_SETTINGS.slidesToShow;
+
+    return (currS + slidesToShow) * moveDistance;
+  }
+}
+
+const calSwipeDirection = (touchObj: { startX: number, startY: number, curX: number, curY: number }): string => {
+  const xDist = touchObj.startX - touchObj.curX;
+  const yDist = touchObj.startY - touchObj.curY;
+  const r = Math.atan2(yDist, xDist);
+
+  let swipeAngle = Math.round(r * 180 / Math.PI);
+
+  if (swipeAngle < 0) {
+    swipeAngle = 360 - Math.abs(swipeAngle);
+  }
+  // swipeAngle between 45 and 135, between 225 and 315 is ignored;
+  if (swipeAngle <= 45 && swipeAngle >= 0 || swipeAngle <= 360 && swipeAngle >= 315) {
```

Comment:
might want to check for a significant change in direction or it will change slide on touch instead of just swipe

Reply:
it's also blocking clicking on a product

Reply:
Changed in another function. Excellent point.